lib/seko_ecom_api/clients/omni_returns_client.rb

```diff
@@ -0,0 +1,60 @@
+require 'faraday'
+require 'faraday/encoding'
+require 'faraday/http'
+require 'active_support/core_ext/hash'
+
+module SekoEcomAPI
+  class OmniReturnsClient
+    include Client
+
+    BASE_URL = 'https://www.omniparcelreturns.com/index.php/api/'.freeze
+    TEST_BASE_URL = 'https://test.omniparcelreturns.com/index.php/api/'.freeze
+
+    attr_reader :access_key, :adapter, :test
+
+    def initialize(access_key:, adapter: :http, test: false)
+      @access_key = access_key
+      @adapter = adapter
+      @test = test
+    end
+
+    def create_shipment(params)
+      response = handle_response post_request('labels/printshipment', params: params)
+      Shipment.new(response)
+    end
+
+    private
+
+    def connection
+      @connection ||= Faraday.new do |conn|
+        conn.url_prefix = test ? TEST_BASE_URL : BASE_URL
+        conn.ssl.verify = test ? false : true
+        unless test
+          conn.options[:open_timeout] = 20 # connection open time out
+          conn.options[:timeout] = 20 # waiting for response time out
+        end
+        conn.headers['Access_Key'] = access_key
+        conn.request :json
+        conn.response :encoding
+        conn.adapter adapter
+      end
+    end
+
+    def handle_response(response)
+      super
+      parsed_response = parse_response(response)
+      response_errors = parsed_response['errors']
+
+      raise(Error, "Something went wrong. #{response_errors}") unless response_errors.empty?
+
+      parsed_response
+    end
+
+    def parse_response(response)
+      parsed_response = JSON.parse(response.body)
+      parsed_response.deep_transform_keys(&:underscore)
+    rescue JSON::ParserError
+      raise Error, 'Response body cannot be parsed. Please refer to the error full message for more details.'
```

Comment:
Please include the response in the Error so that when the user rescue the Error from outside they can do something like this
```rb
rescue SekoEcomAPI::Error => e
  puts e.response.body
end
```